### ISSUE TITLE
Flutterflow Checkbox Tile not functioning as intended

### ISSUE THREAD

**Issue by sushiyade** [FlutterFlow/flutterflow-issues#944]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Checkbox does not show selected behaviours when tapped, duplicates taps are recognised and causes checkbox to add their duplicated values into the widget state of checked items.

This is seen during running on local machine on chrome as well as iOS simulator

### Expected Behavior

What is written in the Flutterflow docs on checkbox list tiles

### Steps to Reproduce

<img width="260" alt="スクリーンショット 2023-05-29 午後2 43 01" src="https://github.com/FlutterFlow/flutterflow-issues/assets/68842066/5d2789ce-d9b6-4b64-9d68-a0f57561f6c1">

### Reproducible from Blank

- [x] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFflcr1z89OoeFF0L6IKMd4qCI4JVMcToAzjdV+bwogIuOqCpkqdcnQVFJuTdO0d1FmPWCKoDsK+dHemvH9FccEGAqYY6phzapIcA3jeGOVabXXCZCJPG5/DeFRN0yg0pmvgCJDCLNsSXAOwznzBdyyRT7YHr60NUQdHv2HJZOK2SrDX1iXc2URm05KZDPz

### Context

behaviour from the docs, create a list of users from selected.

### Visual documentation

https://github.com/FlutterFlow/flutterflow-issues/assets/68842066/bf4e139a-3153-49bd-9793-666a77729bbe

### Environment

```markdown
- FlutterFlow version: 3.7.12
- Platform: Mac OS Ventura 13.1
- Browser name and version: Safari v16.3
- Operating system and version affected: stated above
```

**Comment by hariprasadms** [comment]
Hi @sushiyade - Thanks for submitting the issue. I just checked your project. The page what you are showing in the video does not have checkbox actions configured. So we can't investigate the issue when issue reproducible steps are not configured properly.

Please reproduce the issue in blank project and share the project bug report code please. Thanks

**Comment by sushiyade** [comment]
https://github.com/FlutterFlow/flutterflow-issues/assets/68842066/4df44e81-d2e7-4640-8440-5b1bfcfe3236

ITFflcr1z89OoeFF0L6IKMd4qCI4JVMcToAzjdV+bwogIuOqCpkqdcnQVFJuTdO0T0FEI06j8X0Kp8HemvLtKcInG06WRthzz5ZQbwmXZEGQbL6rDsy8RG5/DeFRN0yg0pmvgCJDCLNsSXAOwznzBdyyRT7YHr60NUQdHv2HJZOK2SrDX1iXc2URm05KZDPz

**Comment by sushiyade** [comment]
There are no actions configured for this checkbox list tile and it does not work as well

**Comment by sushiyade** [comment]
For context, I am trying to get a list of members selected and passing it to another page through a page parameter to create a group with said members.

**Comment by sushiyade** [comment]
This issue occurs when I populate the listview using any query collection, even the ones generated by FlutterFlow without any changes made (chat_messages)

**Comment by younicornsJH** [comment]
I can confirm we experience faulty behaviour on checkboxes as well on several projects of ours in FlutterFlow.
Clicks seem to have no visible effect, however actions are indeed triggered in the background and page reload causes widget update and correct display. 
As this happens on two completely unrelated projects using this component, I think we can rule out any custom code origin here.
Was working perfect before early last week, seems to be broken since then.

**Comment by sushiyade** [comment]
Alright thanks! When will this issue be fixed?

**Comment by younicornsJH** [comment]
I am a user just like you! Better ask @hariprasadms @agreaves I guess...

**Comment by hariprasadms** [comment]
Hi @younicornsJH @sushiyade - I have looked at the given project details above ( According to the bug report code provided above, it appears that the project is not a blank project with the issue setup details ). We are not able investigate the issue as reproducible steps are not clear. 

Please setup a new project with issue setup and provide the project's bug report code. We can investigate the issue once we got the details. 

However, I can confirm that checkbox tile widget is working as expected. Convert the checkbox tile widget as component and build and see if that can help you.

**Comment by younicornsJH** [comment]
Sorry but when reporting bug is more time consuming than fixing the bug itself, there is a problem.
You get 3 reports for 3 different applications which used to work flawlessly, and suddenly the same widget fails on all 3 applications at the same time with the same faulty behaviour. And you stick with "I can confirm that checkbox tile widget is working as expected".
FlutterFlow is really not what it used to be not so long ago.
Anyway, I'm open to a screensharing session or any discussion on this issue.

**Comment by RECPOS** [comment]
Hi @hariprasadms 

try this bug code (page bug code): IT4kj8ncx85Pm9hE1qrtbe4xvGMRC0t9TL07ttYbZxUjIuPtBJh/Os77chBtZMDiTFNhHFKggmoD/sL2i/P+G/ErF0mtG9Q/1c1qehbhUXq8R8m2PoeKTnN9J/9UN36k5rbQ3x5DIuNiSnwl716IAuqRXjHtJ5O/Zwh9f6PHaOI=

Steps to reproduce:
1: Create a blank project
2: Create a Native Cloud database in Firebase and setup all access as the documentation states.
3: create a collection lets call it test_checkbox with field name and selected. 
4: create three documents in the collection, values: {name: "Checkbox 1",selected: false},{name: "Checkbox 2",selected: false},{name: "Checkbox 3",selected: false}
5: Remember to deploy it all to firestore through flutterflow.
6: create a blank page call it "ProductNew"
7: Put a listview (first one) in the column on the page
8: create a backend query with list of documents from the collection test_checkbox, and you can now decide if single time query is needed or not, still gives same issue either way.
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/79512137/c3468b10-7f88-4a03-8cf6-af2f946e5d1b)

9: Put a checkboxlisttile inside the listview, select in Title the name from the backend query data. 
10: Now try run a session

If you like you can also try after step 7 and add a row, then in the row add checkbox and then textfield. Now in the textfiled bind to name, and run again. 

Now if you dont get the same issues as us. then i would really like to know, if there is a difference between the Flutterflow paid packages since the issues appear for us and not you :)

the video for the result. I hope that this demonstrates what does not work at all. 

Video:
https://github.com/FlutterFlow/flutterflow-issues/assets/79512137/a04ef581-edef-4a9e-8ede-0b492546c377

**Comment by Globaldiary2023** [comment]
I'm having exactly the same issue. This set-up flow is perfect

**Comment by Globaldiary2023** [comment]
@hariprasadms your answer is insufficient. Please rectify ASAP

**Comment by Globaldiary2023** [comment]
> 

@sushiyade maybe tick the box on your original message now that there are steps to reproduce?

**Comment by younicornsJH** [comment]
Issue has been ongoing for 1 week now, for such a core component this is really annoying, especially for apps in production.
@agreaves @hariprasadms anything you can do to urge this ?
Many thanks

**Comment by Globaldiary2023** [comment]
See my (similar) [issue](https://github.com/FlutterFlow/flutterflow-issues/issues/952) too

**Comment by hariprasadms** [comment]
Hello @RECPOS - Thanks for sharing the reproducible steps and video details.  I have just tested the give project and checkbox configurations. There are few things are going on. See below points.

1.  You have firebase backend query for list-view widget with checkbox tile widget. 

2. You have document data binding for title of checkbox but not for checkbox status ( initial status of the checkbox ) - 

3. When you click a checkbox in run or debug mode, you typically see nothing happen because the default value of the checkbox is set to "false" ( This is in accordance with the checkbox tile widget parameters). (**Try this**  - Use checkbox actions to update the field's status and bind the checkbox's initial value to the fire-store doc field. Check for the result. Additionally, change the value of the "selected" field to either true or false to observe UI changes. Verify its status.

4. Look at your fire-store rules of test_checkbox collection. It shows as below. Write permission marked as 'No One'. It should be updated and deployed as per project firebase authentication setup.

<img width="1079" alt="image" src="https://github.com/FlutterFlow/flutterflow-issues/assets/17435678/3f3388fe-f0b2-47f0-938b-d318c8c0d63b">

**Note -**  Checkbox tile widget should be used with check status configured properly. I believe that the checkbox and checkbox tile widgets intended purposes are not served if the initial value of the checkbox is incorrect. 

However, when I did try the same steps in my test projects it is working fine. Here is the below video showing checkbox tile working as expected with fire-store doc fields configured correctly ( a text and check status).

https://github.com/FlutterFlow/flutterflow-issues/assets/17435678/826c3a36-63b6-4c04-8cd4-8419c09255be

If an issue still exists in your project after trying the techniques mentioned above, we can mark this as a project-specific issue.

**Comment by hariprasadms** [comment]
Hello All, I hope my suggestions in my last comment is helped to resolve your issues. Please confirm. I will be closing the issue shortly. Thanks

**Comment by sushiyade** [comment]
@hariprasadms Hi could you share more detailed steps to your proposed solution? This has been a blocker for 2 weeks.

**Comment by xray7474** [comment]
I have the same problem also and I do not use Firestore for the query. The only actions set on the checkboxes are to increment or decrement an app state variable. And the checkbox will stop visually working after a few clicks. However the app state keeps incrementing when clicked.

**Comment by agreaves** [comment]
Hi @younicornsJH, apologies in the delay in responding to this issue. I believe this was an issue that was fixed subsequently, is that correct? I've tried reproducing it and have not been able to, as CheckboxList seems to be working properly when populated both from local state and dynamically from Firestore.

Please confirm whether this is the case and I'll close this issue. Thanks!

**Comment by younicornsJH** [comment]
4 months ago I décided to implement my own component as the app was already
in production and this was breaking.
I haven't looked into this matter since then but when I do, I'll update the
status.
Best
JH

Le lun. 18 sept. 2023 à 00:07, Alex Greaves ***@***.***> a
écrit :

**Comment by agreaves** [comment]
Closing as the original issue appears to be fixed. Please re-open or create a new issue if you experience a similar issue!

**Comment by garl99** [comment]
![Screenshot 2024-02-15 203554](https://github.com/FlutterFlow/flutterflow-issues/assets/49343955/c53f91f6-1345-4b6b-b2ca-fc38724046c5)

https://github.com/FlutterFlow/flutterflow-issues/assets/49343955/3d5f241b-c516-43a4-ac87-4da917b68747

Hello, I have the same issue. What is the solution? I'm new using FF

I noticed that when I click on the checkbox the request is resent.

**Comment by mkomarsr** [comment]
Same issue here. The entire page is being reloaded on checkbox click. I have no actions configured on the checkboxes.

**Comment by PMankita** [comment]
I am facing the same issue. Does anyone have any solution?
